scripts/swupd_performance/data_processing_script.py

```diff
@@ -0,0 +1,379 @@
+import os
+from datetime import datetime
+
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+JSON_KEYWORD1 = ["changed bundles", "new bundles", "deleted bundles",
+                 "changed files", "new files", "deleted files",
+                 "Statistics for going from version", "timestamp"]
+
+FILTER1 = ["info"]
+
+STATE_FILE = ".state"
+
+DEFAULT_INPUT_FOLDER = "/var/log/swupd/staging"
+
+# keyword_list3 = ['Total execution time']
+
+
+def santize_get_item(item):
+    """Clean and parse item.
+
+    Clean and parse item of type 'info', split and return key, value.
+
+    Args:
+        item (dict): item to be cleaned
+
+    Returns:
+        a tuple of key, value. returns unknowns strings
+        on Keyerror
+
+    """
+    key = "unknown"
+    value = "unknown"
+    try:
+        key = item['msg'].split(":")[0].strip()
+        value = item['msg'].split(":")[1].strip()
+    except KeyError:
+        pass
+    return key, value
+
+
+def get_to_from_version(stat_string):
+    """Extract to, from version from stat_string.
+
+    Extract the to & from version from a string
+    which has statistics string in it.
+
+    Args:
+        stat_string (string): string to be cleaned
+
+    Returns:
+        a tuple of (from_version, to_version) or 0 is returned for a
+        field whose value cant be determined
+
+    """
+    integers_parsed = [0, 0]
+
+    if len(stat_string) < 3:
+        return integers_parsed
+
+    i = 0
+    for item in stat_string.split():
+        try:
+            integers_parsed[i] = int(item)
+            i = i + 1
+        except ValueError:
+            pass
+    return tuple(integers_parsed)
+
+
+def bundle_stats(bundle_file_stats, outputfile):
+    """Parse a dictionary and writes to outputfile in csv format.
+
+    Parse a dict called bundle_file_stats
+    constructed from the incoming json file and writes it to an outputfile.
+
+    Args:
+        bundle_file_stats (dict): dict to parse values from
+        outputfile (string): the csv file to write to
+
+    """
+    fieldnames = ['changed bundles', 'new bundles',
+                  'deleted bundles', 'changed files',
+                  'new files', 'deleted files',
+                  'from_version', 'to_version', 'timestamp']
+    writer = None
+    bundle_stat_dict = {}
+    file_exist = os.path.isfile(outputfile)
+
+    with open(outputfile, "a") as outfile:
+        writer = csv.DictWriter(
+            outfile,
+            fieldnames=fieldnames,
+            lineterminator='\n')
+
+        if not file_exist:
+            writer.writeheader()
+
+        for keyword in JSON_KEYWORD1:
+            for item in bundle_file_stats:
+                if item['msg'].find(keyword) > -1:
+                    (key, value) = santize_get_item(item)
+                    if keyword == "Statistics for going from version":
+                        (bundle_stat_dict['from_version'],
+                         bundle_stat_dict['to_version']) = get_to_from_version(key)
```

Comment:
I think you can do:
```
                    if keyword == "Statistics for going from version":
                        (bundle_stat_dict['from_version'],
                         bundle_stat_dict['to_version']
                         ) = get_to_from_version(key)
```

Reply:
so you don't have that code style violation

Reply:
It leads to syntax error

Reply:
what about:
```
                    if keyword == "Statistics for going from version":
                        (bundle_stat_dict['from_version'],
                         bundle_stat_dict['to_version']) = \
                            get_to_from_version(key)
```

Reply:
@castulo That worked! I thought I had tried it earlier and didnt. Thanks